.github/workflows/contributed-recipes.yml

```diff
@@ -0,0 +1,47 @@
+name: Test contributed recipes
+
+on:
+  schedule:
+    # Images are rebuilt on Monday, so we're testing recipes each Tuesday
+    # Weekly, at 03:00 on Tuesday UTC time
+    - cron: "0 3 * * 2"
+  pull_request:
+    paths:
+      - ".github/workflows/contributed-recipes.yml"
+      - "docs/using/recipe_code/"
+  push:
+    branches:
+      - main
+    paths:
+      - ".github/workflows/contributed-recipes.yml"
+      - "docs/using/recipe_code/"
```

Comment:
Perfect! I think this makes perfect sense, nice work!

Reply:
@consideRatio

I'm not sure if other projects test their docs, but I really like that this one does, because I had (relatively) many complaints that some recipe was not working.
The coolest thing about this approach is that I can even insert tests (if I want to) to the dockerfiles and not to include them in docs.

https://myst-parser.readthedocs.io/en/latest/syntax/code_and_apis.html#including-code-from-files
There is a really useful option:
```
:end-before: end example
```
So, tests can be any usual RUN/COPY or other docker steps and they will be tested by my workflow, but no one will see them in the docs.

Reply:
Oh thanks for sharing, great idea of using that MyST feature (which I learned about now)!

> many complaints that some recipe was not working.

Yes! I've grown a bit cautious about documentation additions because of such maintenance needs arising. Your work with this is whats required to go hand in hand with growning amounts of recipies I figure!

I think your idea on how to handle this is very promising, I wonder if it can be combined with a badge or note saying when it was last tested and if it was successful at that time or not. For that I think one would need to use some REST API first:
- List runs of a workflow: https://docs.github.com/en/rest/actions/workflow-runs?apiVersion=2022-11-28#list-workflow-runs-for-a-workflow
- List jobs in a workflow run: https://docs.github.com/en/rest/actions/workflow-jobs?apiVersion=2022-11-28#list-jobs-for-a-workflow-run

Hmmm, but that API needs to be done during doc build, which would freeze the info into the docs built HTML and not update over time if it isn't rebuild. So, maybe stashing away the status to be dynamically read from the docs HTML, for example as part of running test jobs or as a job that `needs` the test job (run multiple times) updating some JSON blob to be read from the built documentation directly.

If so, then one could use a badge from shields.io (more about that both at https://shields.io/badges/dynamic-json-badge and https://shields.io/badges/endpoint-badge). As an example of a shields.io based Dynamic JSON badge, see this badge showing the latest version of the jupyterhub Helm chart based on [a json blob](https://hub.jupyter.org/helm-chart/info.json) we update elsewhere:

[![Latest pre-release of the Helm chart](https://img.shields.io/badge/dynamic/json.svg?label=pre&url=https://hub.jupyter.org/helm-chart/info.json&query=$.jupyterhub.pre&colorB=orange&logo=helm)](https://jupyterhub.github.io/helm-chart#development-releases-jupyterhub)

```markdown
[![Latest pre-release of the Helm chart](https://img.shields.io/badge/dynamic/json.svg?label=pre&url=https://hub.jupyter.org/helm-chart/info.json&query=$.jupyterhub.pre&colorB=orange&logo=helm)](https://jupyterhub.github.io/helm-chart#development-releases-jupyterhub)
```

Reply:
I will think about this, I don't have the answer to this currently.

> Hmmm, but that API needs to be done during doc build, which would freeze the info into the docs built HTML and not update over time if it isn't rebuild.

I might have a solution to this problem, but it's not for everyone, I suppose.
Use scheduled builds.
Yes, you will have more builds (so, more CPU spent), more failing builds and will need to add a few lines of code to GitHub Workflows. But at the same time, you will catch bugs earlier, if they happen - and they are much easier to track down, when you know it started occurring only in the last week, not since the previous build.
I suppose it's easy to add a badge with the date of build.
And it's easy to add a badge that the GitHub workflow was last successful or not.